code/game/machinery/computer/sentencing.dm

```diff
@@ -1,481 +1,266 @@
-#define IMPORT_INCIDENT "import_incident"
 #define PERMABRIG_SENTENCE 60 // Measured in minutes
+
 /obj/structure/machinery/computer/sentencing
 	name = "\improper Jurisdictional Automated System"
 	desc = "A powerful machine produced by Weyland-Yutani to streamline all punishment of prisoners. The best grade policing gear seen on this side of the galaxy."
 	icon_state = "jas"
 	req_one_access = list(ACCESS_MARINE_BRIG, ACCESS_MARINE_BRIDGE)
-
 	var/datum/crime_incident/incident
-	var/menu_screen = "main_menu"
-	var/paper_counter = 0
-	var/imported_paper = FALSE
+	var/current_menu = "main"
+	var/static/paper_counter = 0
 
 /obj/structure/machinery/computer/sentencing/attack_hand(mob/user as mob)
 	if(..() || !allowed(usr) || inoperable())
 		return
 
-	ui_interact(user)
+	tgui_interact(user)
 
-/obj/structure/machinery/computer/sentencing/attackby(var/obj/item/O, var/mob/user)
-	if(istype(O, /obj/item/paper/incident) && menu_screen == IMPORT_INCIDENT)
-		user.temp_drop_inv_item(O)
-		O.forceMove(loc)
+/obj/structure/machinery/computer/sentencing/tgui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "Sentencing", "Criminal Sentencing")
+		ui.open()
+		ui.set_autoupdate(FALSE)
 
-		if(import(O))
-			ping("\The [src] pings, \"Successfully imported incident report!\"")
-			menu_screen = "incident_report"
-			updateUsrDialog()
-		else
-			to_chat(user, SPAN_ALERT("Could not import incident report."))
+/obj/structure/machinery/computer/sentencing/ui_data(mob/user)
+	. = list()
```

Comment:
I will not force you to change this but typically we discourage `.`-hijacking for ui_data